src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+                for entry in entry_path.iterdir():
+                    if entry.is_symlink():
+                        continue
+                    if entry.is_file() and entry.name.endswith(".yaml"):
+                        yield entry
+            elif entry_path.is_file():
+                yield entry_path
+
+    for file_path in _iterate_files_or_dirs():
+        try:
+            with resolve_pipeline(file_path) as pipeline_file:
+                yield str(file_path), str(pipeline_file)
+        except NotAPipelineFile as e:
+            logger.warning("%s is not an expected pipeline file due to: %s", file_path.name, e)
+        except Exception as e:
+            logger.warning("%s seems not a YAML file due to: %s", file_path.name, e)
+
+
+class KonfluxBuildDefinitions:
+
+    DEFINITIONS_REPO: Final = "konflux-ci/build-definitions"
+    KONFLUX_IMAGE_ORG: Final = "konflux-ci/tekton-catalog"
+    VERSION_REGEX: Final = re.compile(r"^(\d+)\.(\d+)$")
+
+    @staticmethod
+    def extract_task_names(tasks: list[dict[str, Any]]) -> Generator[tuple[str, str]]:
+        """Extract pipeline task name and actual task name from a task list
+
+        :return: a generator that yields a list of two-elements tuples. The first one is the
+            pipeline task name, and the second one is the actual task name.
+        """
+        for t in tasks:
+            pipeline_task_name = t.get("name")
+            if not pipeline_task_name:
+                logger.warning("Cannot get pipeline task name from %r, skip it:", t)
+                continue
+            task_ref = t.get("taskRef")
+            if not task_ref:
+                logger.warning("Task %s does not have taskRef. Skip it.", pipeline_task_name)
+                continue
+            if task_ref.get("resolver") != "bundles":
+                logger.warning("Task %s does not use tekton bundle. Skip it.", pipeline_task_name)
+                continue
+            actual_task_name = None
+            for param in task_ref["params"]:
+                if param["name"] == "name":
+                    actual_task_name = param["value"]
+                    break
+            if not actual_task_name:
+                logger.warning(
+                    "Task %s uses tekton bundle resolver but no actual task name is specified "
+                    "in the resolver. Skip it.",
+                    pipeline_task_name,
+                )
+                continue
+            yield pipeline_task_name, actual_task_name
+
+    @staticmethod
+    def validate_bundle_ref(bundle_ref: str) -> None:
+        try:
+            c = Container(bundle_ref)
+        except ValueError as e:
+            # The underlying oras Container.parse raises ValueError
+            raise ValueError(f"{bundle_ref} is not a valid image reference: {str(e)}")
+        if c.registry == oras.defaults.registry.index_name:
+            raise ValueError("Currently only support adding Konflux tasks from quay.io.")
```

Comment:
Based on the error message, should the condition be `if c.registry != "quay.io"`?